tests/e2e/system_test/test_full_cluster_health.py

```diff
@@ -0,0 +1,286 @@
+            run_command(f"oc -n {benchmark_operator.BMO_NAME} delete pvc {pvc}")
+
+        # Getting all PVs created in the test (if left).
+        command[2] = "pv"
+        command[8] = (
+            "'{{range .items}}{{.metadata.name}} {{.spec.claimRef.namespace}}{{\""
+            + NL
+            + "\"}}{{end}}'"
+        )
+        command.remove("-n")
+        command.remove(benchmark_operator.BMO_NAME)
+        pvs_list = run_command(command, out_format="list")
+        logger.info(f"list of all PVs :{pvs_list}")
+
+        for line in pvs_list:
+            try:
+                pv, ns = line.split(" ")
+                pv = pv.replace("'", "")
+                if ns == benchmark_operator.BMO_NAME:
+                    logger.info(f"Going to delete {pv}")
+                    run_command(f"oc delete pv {pv}")
+            except Exception:
+                pass
+
+    def delete_pods(self):
+        """
+        Try to delete pods:
+            - Rook operator
+            - OSD
+            - MGR
+            - MON
+        """
+        pod_list = []
+        rook_operator_pod = pod.get_ocs_operator_pod(
+            ocs_label=constants.OPERATOR_LABEL,
+            namespace=constants.OPENSHIFT_STORAGE_NAMESPACE,
+        )
+        pod_list.append(rook_operator_pod)
+
+        osd_pods = pod.get_osd_pods()
+        pod_list.extend(osd_pods)
+
+        mgr_pods = pod.get_mgr_pods()
+        pod_list.extend(mgr_pods)
+
+        mon_pods = pod.get_mon_pods()
+        pod_list.extend(mon_pods)
+
+        logger.info(f"Deleting pods: {[p.name for p in pod_list]}")
+        pod.delete_pods(pod_objs=pod_list)
+
+    def ceph_not_health_error(self):
+        """
+        Check if Ceph is NOT in "HEALTH_ERR" state
+        Warning state is ok since the cluster is low in storage space
+
+        Returns:
+            bool: True if Ceph state is NOT "HEALTH_ERR"
+        """
+        ceph_status = self.ceph_cluster.get_ceph_health()
+        logger.info(f"Ceph status is: {ceph_status}")
+        return ceph_status != "HEALTH_ERR"
+
+    def mgr_pod_node_restart(self):
+        """
+        Restart node that runs mgr pod
+        """
+
+        mgr_pod_obj = pod.get_mgr_pods()
+        mgr_node_obj = pod.get_pod_node(mgr_pod_obj[0])
+
+        self.nodes.restart_nodes([mgr_node_obj])
+
+        wait_for_nodes_status()
+
+        # Check for Ceph pods
+        pod_obj = ocp.OCP(kind=constants.POD, namespace=defaults.ROOK_CLUSTER_NAMESPACE)
+        assert pod_obj.wait_for_resource(
+            condition="Running", selector="app=rook-ceph-mgr", timeout=600
+        )
+        assert pod_obj.wait_for_resource(
+            condition="Running",
+            selector="app=rook-ceph-mon",
+            resource_count=3,
+            timeout=600,
+        )
+        assert pod_obj.wait_for_resource(
+            condition="Running",
+            selector="app=rook-ceph-osd",
+            resource_count=3,
+            timeout=600,
+        )
```

Comment:
This can be removed. After the expected mon are in running state, you are checking is_cluster_healthy and that should cover osd pods check.